src/analyzer/protocol/login/Login.cc

```diff
@@ -562,6 +562,7 @@ void Login_Analyzer::AddUserText(const char* line)
 		if ( ++user_text_last == MAX_USER_TEXT )
 			user_text_last = 0;
 
+		delete[] user_text[user_text_last];
```

Comment:
I don't think this is correct.  There's the `num_user_text >= MAX_USER_TEXT` check above that doesn't allow the `user_text` capacity to be exceeded or wrap such that we'd ever replace an entry that still points to memory in need of being freed.  Instead, it seems all calls to `PopUserText()` are responsible for freeing the returned entry (or else transferring ownership), so adding the `delete[]` here would cause a double-free if it eventually does wrap and start re-using previous indices.

Reply:
You're right, the commit is wrong!
I was confused by the constructor which explicitly initializes all `user_text` elements to `nullptr`. This overhead makes no sense if dangling pointers are allowed.